pkg/scheduler/framework/status.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package framework
+
+import "strings"
+
+// StatusCode is the status code of a Status, returned by a plugin.
+type StatusCode int
+
+// Pre-defined status codes.
+const (
+	// Success signals that a plugin has completed its run successfully.
+	// Note that a nil *Status is also considered as a Success Status.
+	Success StatusCode = iota
+	// internalError signals that a plugin has encountered an internal error.
+	// Note that this status code is NOT exported; to return an internalError status, use the
+	// FromError() call.
+	internalError
+	// ClusterUnschedulable signals that a plugin has found that a placement should not be bound
+	// to a specific cluster.
+	ClusterUnschedulable
+	// Skip signals that no action is needed for the plugin to take at the stage.
+	// If this is returned by a plugin at the Pre- stages (PreFilter or PreScore), the associated
```

Comment:
we have PreFilter/PreScore?

Reply:
Yeah, this probably is better explained in the following interface PR, but the general idea is that we need the pre- stages so that some plugins can determine that they do not need to be called early (i.e., if there is no topology spread constraints, there is no need to call topology spread constraints plugin at all), otherwise per our interface these plugins will be called once per cluster ->  inspect the policy -> then terminate its call, which incurs some extra overhead.